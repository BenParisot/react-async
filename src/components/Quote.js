import React from 'react';
import PropTypes from 'prop-types';
import Styles from '../main.css';

export default function Quote({ character, quote, image }) {
  return (
    <ul className={Styles.quote}>
      <li>{character}</li>
      <li>{quote}</li>
      <li>{image}</li>
    </ul>
  );
}
Quote.propTypes = {
  character: PropTypes.string.isRequired,
  quote: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired
};
