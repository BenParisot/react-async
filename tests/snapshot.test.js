import React from 'react';
import { shallow } from 'enzyme';
import Quote from '../src/components/Quote';

describe('Quote component', () => {
  it('displays a quote', () => {
    const wrapper = shallow(<Quote character='Bender' quote='Bite my shiny metal ass!' image='img/bender.png' />);
    expect(wrapper).toMatchSnapshot();
  });
});
