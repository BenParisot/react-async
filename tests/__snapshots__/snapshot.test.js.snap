// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character component displays a character 1`] = `
<ul
  className="quote"
>
  <li>
    Name: 
    Rick
  </li>
  <li>
    Status: 
    alive
  </li>
  <li>
    Species: 
    human
  </li>
  <li>
    img/rick.png
  </li>
</ul>
`;
